File: 0804-unique-morse-code-words/0804-unique-morse-code-words.cs
public class Solution {
    public int UniqueMorseRepresentations(string[] words) {
        String[] morseCodes = {".-", "-...", "-.-.", "-..", ".", "..-.", "--.", "....",
                              "..", ".---", "-.-", ".-..", "--", "-.", "---", ".--.",
                              "--.-", ".-.", "...", "-", "..-", "...-", ".--", "-..-",
                              "-.--", "--.."};
        List<String> convList = new List<String>();
        for (int i = 0; i < words.Length; i++) {
            char[] charsInWord = words[i].ToCharArray();
            String converted = "";
            
            for (int j = 0; j < charsInWord.Length; j++) {
                converted += morseCodes[charsInWord[j] - 'a'];
            }
            convList.Add(converted);
        }
        convList = convList.Distinct().ToList();
        return convList.Count;
    }
}

File: 0804-unique-morse-code-words/0804-unique-morse-code-words.java
class Solution {
    public int uniqueMorseRepresentations(String[] words) {
        String[] morseCodes = {".-", "-...", "-.-.", "-..", ".", "..-.", "--.", "....",
                              "..", ".---", "-.-", ".-..", "--", "-.", "---", ".--.",
                              "--.-", ".-.", "...", "-", "..-", "...-", ".--", "-..-",
                              "-.--", "--.."};
        HashSet<String> convertedSet = new HashSet<String>();
        for (int i = 0; i < words.length; i++) {
            char[] charsInWord = words[i].toCharArray();
            String converted = "";
            
            for (int j = 0; j < charsInWord.length; j++) {
                converted += morseCodes[charsInWord[j] - 'a'];
            }
            convertedSet.add(converted);
        }
        return convertedSet.size();
    }
}